src/app/dialog/dialog.service.ts

```diff
@@ -0,0 +1,26 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import { ConfirmDialogComponent } from './confirm-dialog/confirm-dialog.component';
+import { MatDialogRef, MatDialog } from '@angular/material';
+
+@Injectable()
+export class DialogService {
+
+    constructor(private dialog: MatDialog) { }
+
+    public confirm(title: string, message: string): Observable<boolean> {
+
+        let dialogRef: MatDialogRef<ConfirmDialogComponent>;
+
+        dialogRef = this.dialog.open(ConfirmDialogComponent, {
+          width: '400px',
+          hasBackdrop: true,
+          panelClass: 'centered-dialog',
+        });
+
+        dialogRef.componentInstance.title = title;
```

Comment:
Is this the only way to access the instance?

Reply:
Another question is: do you even need `ConfirmDialogComponent` or just a `TemplateRef`?